package com.jpmc.midascore;

import org.springframework.kafka.core.KafkaTemplate;
import org.springframework.stereotype.Component;
import com.jpmc.midascore.foundation.Transaction;  // YEH IMPORT ADD KARO

@Component
public class KafkaProducer {

    private final KafkaTemplate<String, Transaction> kafkaTemplate;

    public KafkaProducer(KafkaTemplate<String, Transaction> kafkaTemplate) {
        this.kafkaTemplate = kafkaTemplate;
    }

    public void send(String topic, Transaction transaction) {
        kafkaTemplate.send(topic, transaction);
    }
}